File: dotnet/Multiplayer/Matchmaking/Requests/MatchmakingPermissionRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Matchmaking.Requests
{
    /// <summary>
    /// Request data for responding to a matchmaking join request with approve/reject action.
    /// </summary>
    public class MatchmakingPermissionRequest
    {
        [JsonInclude]
        private string Action { get; set; } = EMatchmakingRequestAction.Approve.ToString().ToLower();

        /// <summary>
        /// Initializes a new MatchmakingPermissionRequest.
        /// </summary>
        /// <param name="action">The action to take (Approve or Reject).</param>
        public MatchmakingPermissionRequest(EMatchmakingRequestAction action)
        {
            Action = action.ToString().ToLower();
        }
    }
}

File: java/src/main/java/com/michitai/multiplayer/matchmaking/requests/MatchmakingPermissionRequest.java
package com.michitai.multiplayer.matchmaking.requests;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Request for responding to a matchmaking join request with approve/reject action.
 */
public class MatchmakingPermissionRequest {
    @JsonProperty("action")
    private String action;

    public MatchmakingPermissionRequest(String action) {
        this.action = action;
    }

    public String getAction() {
        return action;
    }

    public void setAction(String action) {
        this.action = action;
    }
}
